Sources/SwiftNetKit/BaseRequest.swift

```diff
@@ -46,7 +46,15 @@ public struct BaseRequest<Response: Decodable>: RequestProtocol {
         urlRequest.allHTTPHeaderFields = self.headers
         
         if let body = self.body {
-            urlRequest.httpBody = body
+            switch body {
+            case .data(let data):
+                urlRequest.httpBody = data
+            case .string(let string):
+                urlRequest.httpBody = string.data(using: .utf8)
+            case .jsonEncodable(let encodable):
+                let jsonData = try? JSONEncoder().encode(encodable)
+                urlRequest.httpBody = jsonData
+            }
```

Comment:
You could consider adding some code in here as well to automatically add `Content-Type` header to the urlRequest depending on the body type if the user doesn't manually add it